Bug 1806028: IPv6 support for inspector

stbenjam opened the pull request:
This commit makes the neccessary changes for Inspector to work on IPv6,
including handling brakcets around URL's, and using the PROVISIONING_IP
variable to wait for a known IP on an interface, which may be an IPv6
address.

Cherry-picks of:
 - metal3-io#47
 - metal3-io#42
 - metal3-io#43

4.5 BZ 1806028
4.4 BZ 1806029
4.3 BZ 1806032

stbenjam committed:
IPv6 support for inspector

This commit makes the neccessary changes for Inspector to work on IPv6,
including handling brakcets around URL's, and using the PROVISIONING_IP
variable to wait for a known IP on an interface, which may be an IPv6
address.

stbenjam committed:
Sync ironic-common.sh with ironic-image

The inspector image has a typo in ironic-common.sh that's preventing the
endpoint for Ironic to be set correctly when using IPv6.  This matches
ironic-common.sh with the exact one from the ironic-image which already
got the fix for this.

stbenjam commented:
/cherry-pick release-4.4

stbenjam commented:
/cherry-pick release-4.3

stbenjam commented:
/assign @derekhiggins

derekhiggins commented on ironic-common.sh:
```diff
@@ -0,0 +1,29 @@
+    else
+      IRONIC_URL_HOST=$IRONIC_IP
+    fi
+  fi
```
Travelling at the moment so didn't have a change to test but looks like this if/else should end before we check $IRONIC_IP for IP version?

stbenjam replied:
You are right! There was already a fix for this IIRC. Will cherry-pick that here.

stbenjam replied:
Fixed

hardys replied:
Sounds like this ended up more than one cherry-pick, should we reference e.g https://github.com/metal3-io/ironic-inspector-image/pull/42 and any other PRs in the commit message?

stbenjam replied:
Sure, fixed.

stbenjam committed:
Sync ironic-common.sh from ironic-image

This syncs the copy of ironic-common.sh with the same one that's in
ironic-image.

Cherry-picks of:
 - https://github.com/metal3-io/ironic-inspector-image/pull/47
 - https://github.com/metal3-io/ironic-inspector-image/pull/42
 - https://github.com/metal3-io/ironic-inspector-image/pull/43

hardys commented:
/approve

hardys commented:
/lgtm
